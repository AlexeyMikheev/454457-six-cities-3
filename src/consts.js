
export const Feature = {
  WIFI: `Wi-Fi`,
  WASHINGMACHINE: `Washing machine`,
  TOWELS: `Towels`,
  HEATING: `Heating`,
  COFFEEMACHINE: `Coffee machine`,
  BABYSET: `Baby seat`,
  KITCHEN: `Kitchen`,
  DISHWASHER: `Dishwasher`,
  CABELTV: `Cabel TV`,
  FRIDGE: `Fridge`
};

export const FEATURES = [
  Feature.WIFI,
  Feature.WASHINGMACHINE,
  Feature.HEATING,
  Feature.COFFEEMACHINE,
  Feature.BABYSET,
  Feature.DISHWASHER,
  Feature.CABELTV,
  Feature.FRIDGE
];

export const OfferType = {
  APARTMENT: `Apartment`,
  ROOM: `Private room`,
  HOUSE: `House`,
  HOTEL: `Hotel`
};

export const OFFERTYPES = [
  OfferType.APARTMENT,
  OfferType.ROOM,
  OfferType.HOUSE,
  OfferType.HOTEL
];

export const ViewMode = {
  Main: 0,
  Property: 1
};

export const VIEWMODES = [
  ViewMode.Main,
  ViewMode.Property
];

export const MIN_RATING = 1;
export const MAX_RATING = 5;

export const MAX_REVIEWS_DISPLAY_COUNT = 10;

export const MAX_IMAGES_DISPLAY_COUNT = 6;
export const MAX_NEAR_DISPLAY_COUNT = 3;

export const SortType = {
  POPULAR: `Popular`,
  PRICE_LH: `Price: low to high`,
  PRICE_HL: `Price: high to low`,
  TOPRATED: `Top rated first`
};

export const SORTTYPES = [
  SortType.POPULAR,
  SortType.PRICE_LH,
  SortType.PRICE_HL,
  SortType.TOPRATED
];

export const RatingValues = [5, 4, 3, 2, 1];

export const ErrorType = {
  UNAUTHORIZED: 401,
  BABREQUEST: 400
};

export const TIMEOUT = 5000;
export const URL = `https://htmlacademy-react-3.appspot.com/six-cities`;

export const AuthStatus = {
  NO_AUTH: `NO_AUTH`,
  AUTH: `AUTH`
};

export const LoadingStatus = {
  DEFAULT: `DEFAULT`,
  SUCCESS: `SUCCESS`,
  LOADING: `LOADING`,
  ERROR: `ERROR`
};

export const LoadingStatuses = [
  LoadingStatus.DEFAULT,
  LoadingStatus.SUCCESS,
  LoadingStatus.LOADING,
  LoadingStatus.ERROR
];

export const Url = {
  COMMENTS: `comments`,
  HOTELS: `hotels`,
  LOGIN: `login`,
  NEARBY: `nearby`
};

export const AppRoute = {
  LOGIN: `/login`,
  FAVORITES: `/favorites`,
  OFFER: `/offer`,
  ROOT: `/`,
};
