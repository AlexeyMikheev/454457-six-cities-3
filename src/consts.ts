export enum Feature {
    WIFI= `Wi-Fi`,
    WASHINGMACHINE= `Washing machine`,
    TOWELS= `Towels`,
    HEATING= `Heating`,
    COFFEEMACHINE= `Coffee machine`,
    BABYSET= `Baby seat`,
    KITCHEN= `Kitchen`,
    DISHWASHER= `Dishwasher`,
    CABELTV= `Cabel TV`,
    FRIDGE= `Fridge`
  }

export const FEATURES: Feature[] = [
  Feature.WIFI,
  Feature.WASHINGMACHINE,
  Feature.HEATING,
  Feature.COFFEEMACHINE,
  Feature.BABYSET,
  Feature.DISHWASHER,
  Feature.CABELTV,
  Feature.FRIDGE
];

export enum OfferType {
  APARTMENT= `Apartment`,
  ROOM= `Private room`,
  HOUSE= `House`,
  HOTEL= `Hotel`
}

export const OFFERTYPES: OfferType[] = [
  OfferType.APARTMENT,
  OfferType.ROOM,
  OfferType.HOUSE,
  OfferType.HOTEL
];

export enum ViewMode {
  Main= 0,
  Property= 1,
  Favorite= 2
}

// export const VIEWMODES: ViewMode[] = [
//   ViewMode.Main,
//   ViewMode.Property,
//   ViewMode.Favorite
// ];

export const MIN_RATING = 1;
export const MAX_RATING = 5;

export const MAX_REVIEWS_DISPLAY_COUNT = 10;

export const MAX_IMAGES_DISPLAY_COUNT = 6;
export const MAX_NEAR_DISPLAY_COUNT = 3;

export enum SortType {
  POPULAR= `Popular`,
  PRICE_LH= `Price: low to high`,
  PRICE_HL= `Price: high to low`,
  TOPRATED= `Top rated first`
}

export const SORTTYPES: SortType[] = [
  SortType.POPULAR,
  SortType.PRICE_LH,
  SortType.PRICE_HL,
  SortType.TOPRATED
];

export const RatingValues = [`5`, `4`, `3`, `2`, `1`];

export enum ErrorType {
  UNAUTHORIZED= 401,
  BABREQUEST= 400,
  NOTFOUND= 404
}

export const TIMEOUT = 5000;
export const URL = `https://htmlacademy-react-3.appspot.com/six-cities`;

export enum AuthStatus {
    NO_AUTH= `NO_AUTH`,
    LOADING= `LOADING`,
    AUTH= `AUTH`
  }

export enum LoadingStatus {
  DEFAULT= `DEFAULT`,
  SUCCESS= `SUCCESS`,
  LOADING= `LOADING`,
  ERROR= `ERROR`
}

export const LoadingStatuses: LoadingStatus[] = [
  LoadingStatus.DEFAULT,
  LoadingStatus.SUCCESS,
  LoadingStatus.LOADING,
  LoadingStatus.ERROR
];

export enum Url {
  COMMENTS= `comments`,
  HOTELS= `hotels`,
  FAVORITE= `favorite`,
  LOGIN= `login`,
  NEARBY= `nearby`
}

export enum AppRoute {
  LOGIN= `/login`,
  FAVORITES= `/favorites`,
  OFFER= `/offer`,
  ROOT= `/`,
}


export enum FavoriteState {
  MARKED= 1,
  UNMARKED= 0
}

export enum FavoriteButtonType {
  PROPERTY= 1,
  CARD= 0
}

// export const FavoriteButtonTypes = [
//   FavoriteButtonType.PROPERTY,
//   FavoriteButtonType.CARD
// ];

export enum ImageCardWidth {
  CARD_WIDTH= 260,
  CARD_HEIGHT= 200,
  FAVORITE_WIDTH= 150,
  FAVORITE_HEIGHT= 110
}

export const MIN_COMMENT_LENGTH = 50;
