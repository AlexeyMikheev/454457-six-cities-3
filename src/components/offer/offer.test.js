import React from "react";
import renderer from "react-test-renderer";
import Offer from "./offer.jsx";
import {OfferType} from '../../consts.js';

const mock = {
  id: 1,
  isPremium: true,
  cost: 120,
  isMarked: false,
  rating: 4,
  name: `Beautiful & luxurious apartment at great location`,
  type: OfferType.APARTMENT,
  image: `img/apartment-01.jpg`
};

it(`Render App`, () => {

  const tree = renderer
    .create(<Offer offer={mock} onPlaceHeaderClick={() => {}} onPlaceCardMouseEnter={() => {}} onPlaceCardMouseLeave={() => {}} />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
