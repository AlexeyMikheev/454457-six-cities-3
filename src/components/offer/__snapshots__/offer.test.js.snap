// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Offer snapshot (ViewMode.Favorite) 1`] = `
<article
  class="place-card favorites__card"
>
  <div
    class="place-card__image-wrapper favorites__image-wrapper"
  >
    <a
      xlink:href="#"
    >
      <img
        alt="Place image"
        class="place-card__image"
        height="110"
        src="img/apartment-01.jpg"
        width="150"
      />
    </a>
  </div>
  <div
    class="place-card__info favorites__card-info"
  >
    <div
      class="place-card__price-wrapper"
    >
      <div
        class="place-card__price"
      >
        <b
          class="place-card__price-value"
        >
          €
          120
        </b>
        <span
          class="place-card__price-text"
        >
          / night
        </span>
      </div>
      <button
        class="place-card__bookmark-button button "
        type="button"
      >
        <svg
          class="place-card__bookmark-icon"
          height="19"
          width="18"
        >
          <use
            xlink:href="#icon-bookmark"
          />
        </svg>
        <span
          class="visually-hidden"
        >
          To bookmarks
        </span>
      </button>
    </div>
    <div
      class="place-card__rating rating"
    >
      <div
        class="place-card__stars rating__stars"
      >
        <span
          style="width: 80%;"
        />
        <span
          class="visually-hidden"
        >
          Rating
        </span>
      </div>
    </div>
    <h2
      class="place-card__name"
    >
      <a
        href="/offer/1"
      >
        Beautiful & luxurious apartment at great location
      </a>
    </h2>
    <p
      class="place-card__type"
    >
      Apartment
    </p>
  </div>
</article>
`;

exports[`Offer snapshot (ViewMode.Main) 1`] = `
<article
  class="place-card cities__place-card"
>
  <div
    class="place-card__mark"
  >
    <span>
      Premium
    </span>
  </div>
  <div
    class="place-card__image-wrapper cities__image-wrapper"
  >
    <a
      xlink:href="#"
    >
      <img
        alt="Place image"
        class="place-card__image"
        height="200"
        src="img/apartment-01.jpg"
        width="260"
      />
    </a>
  </div>
  <div
    class="place-card__info "
  >
    <div
      class="place-card__price-wrapper"
    >
      <div
        class="place-card__price"
      >
        <b
          class="place-card__price-value"
        >
          €
          120
        </b>
        <span
          class="place-card__price-text"
        >
          / night
        </span>
      </div>
      <button
        class="place-card__bookmark-button button "
        type="button"
      >
        <svg
          class="place-card__bookmark-icon"
          height="19"
          width="18"
        >
          <use
            xlink:href="#icon-bookmark"
          />
        </svg>
        <span
          class="visually-hidden"
        >
          To bookmarks
        </span>
      </button>
    </div>
    <div
      class="place-card__rating rating"
    >
      <div
        class="place-card__stars rating__stars"
      >
        <span
          style="width: 80%;"
        />
        <span
          class="visually-hidden"
        >
          Rating
        </span>
      </div>
    </div>
    <h2
      class="place-card__name"
    >
      <a
        href="/offer/1"
      >
        Beautiful & luxurious apartment at great location
      </a>
    </h2>
    <p
      class="place-card__type"
    >
      Apartment
    </p>
  </div>
</article>
`;

exports[`Offer snapshot (ViewMode.Property) 1`] = `
<article
  class="place-card near-places__card"
>
  <div
    class="place-card__mark"
  >
    <span>
      Premium
    </span>
  </div>
  <div
    class="place-card__image-wrapper near-places__image-wrapper"
  >
    <a
      xlink:href="#"
    >
      <img
        alt="Place image"
        class="place-card__image"
        height="200"
        src="img/apartment-01.jpg"
        width="260"
      />
    </a>
  </div>
  <div
    class="place-card__info "
  >
    <div
      class="place-card__price-wrapper"
    >
      <div
        class="place-card__price"
      >
        <b
          class="place-card__price-value"
        >
          €
          120
        </b>
        <span
          class="place-card__price-text"
        >
          / night
        </span>
      </div>
      <button
        class="place-card__bookmark-button button "
        type="button"
      >
        <svg
          class="place-card__bookmark-icon"
          height="19"
          width="18"
        >
          <use
            xlink:href="#icon-bookmark"
          />
        </svg>
        <span
          class="visually-hidden"
        >
          To bookmarks
        </span>
      </button>
    </div>
    <div
      class="place-card__rating rating"
    >
      <div
        class="place-card__stars rating__stars"
      >
        <span
          style="width: 80%;"
        />
        <span
          class="visually-hidden"
        >
          Rating
        </span>
      </div>
    </div>
    <h2
      class="place-card__name"
    >
      <a
        href="/offer/1"
      >
        Beautiful & luxurious apartment at great location
      </a>
    </h2>
    <p
      class="place-card__type"
    >
      Apartment
    </p>
  </div>
</article>
`;
